internal/provider/data_source_flink_compute_pool.go

```diff
@@ -125,18 +125,18 @@ func computePoolDataSourceReadUsingId(ctx context.Context, d *schema.ResourceDat
 	tflog.Debug(ctx, fmt.Sprintf("Reading Flink Compute Pool %q=%q", paramId, computePoolId), map[string]interface{}{computePoolLoggingKey: computePoolId})
 
 	c := meta.(*Client)
-	computePool, _, err := executeComputePoolRead(c.fcpmApiContext(ctx), c, environmentId, computePoolId)
+	computePool, resp, err := executeComputePoolRead(c.fcpmApiContext(ctx), c, environmentId, computePoolId)
 	if err != nil {
-		return diag.Errorf("error reading Flink Compute Pool %q: %s", computePoolId, createDescriptiveError(err))
+		return diag.Errorf("error reading Flink Compute Pool %q: %s", computePoolId, createDescriptiveError(err, resp))
 	}
 	computePoolJson, err := json.Marshal(computePool)
 	if err != nil {
-		return diag.Errorf("error reading Flink Compute Pool %q: error marshaling %#v to json: %s", computePoolId, computePool, createDescriptiveError(err))
+		return diag.Errorf("error reading Flink Compute Pool %q: error marshaling %#v to json: %s", computePoolId, computePool, createDescriptiveError(err, resp))
 	}
 	tflog.Debug(ctx, fmt.Sprintf("Fetched Flink Compute Pool %q: %s", computePoolId, computePoolJson), map[string]interface{}{computePoolLoggingKey: computePoolId})
 
 	if _, err := setComputePoolAttributes(d, computePool); err != nil {
-		return diag.FromErr(createDescriptiveError(err))
+		return diag.FromErr(createDescriptiveError(err, resp))
```

Comment:
`resp` may have an outdated value here, shall we keep `diag.FromErr(createDescriptiveError(err))`?

Reply:
Well it depends, but resp has got an error code, it should have exited the code earlier 😁 

It feels like this will be a no-op, but in a hypothetical scenario, we don't want to mix JSON errors with a random API response.
```
computePoolJson, err := json.Marshal(computePool)
	if err != nil {
		return diag.Errorf("error reading Flink Compute Pool %q: error marshaling %#v to json: %s", computePoolId, computePool, createDescriptiveError(err, resp))
```